src/tv2-common/actions/executeAction.ts

```diff
@@ -676,27 +696,46 @@ async function cutServerToBox<
 			!existingCasparObj.metaData.mediaPlayerSession
 		) {
 			context.notifyUserWarning(`Failed to start DVE with server`)
-			return dvePiece
+			return newDvePiece
 		}
 
-		const ssrcObj = (dvePiece.content.timelineObjects as Array<TSR.TSRTimelineObj & TimelineBlueprintExt>)[ssrcObjIndex]
+		const ssrcObj = (newDvePiece.content.timelineObjects as Array<TSR.TSRTimelineObj & TimelineBlueprintExt>)[
+			ssrcObjIndex
+		]
 
 		ssrcObj.metaData = {
 			...ssrcObj.metaData,
 			mediaPlayerSession: existingCasparObj.metaData.mediaPlayerSession
 		}
 
-		dvePiece.content.timelineObjects[ssrcObjIndex] = ssrcObj
-		dvePiece.content.timelineObjects.push(EnableServer(existingCasparObj.metaData.mediaPlayerSession))
+		newDvePiece.content.timelineObjects[ssrcObjIndex] = ssrcObj
+		newDvePiece.content.timelineObjects.push(EnableServer(existingCasparObj.metaData.mediaPlayerSession))
+		;(newDvePiece.metaData as any).mediaPlayerSessions = [existingCasparObj.metaData.mediaPlayerSession]
 
-		if (!dvePiece.metaData) {
-			dvePiece.metaData = {}
+		if (!containedServerBefore) {
+			startServerMetadata(context, meta, modifiesCurrent)
 		}
+	}
+
+	return newDvePiece
+}
 
-		;(dvePiece.metaData as any).mediaPlayerSessions = [existingCasparObj.metaData.mediaPlayerSession]
+function stopServerMetadata(context: ITV2ActionExecutionContext, metaData: DVEPieceMetaData) {
+	const length = metaData.serverPlaybackTiming?.length
+	if (metaData.serverPlaybackTiming && length) {
+		metaData.serverPlaybackTiming[length - 1].end = context.getCurrentTime()
 	}
+}
 
-	return dvePiece
+function startServerMetadata(
```

Comment:
Minor thing:
Should capitalize the 'd' in metaData. It is all the other places.

Comment also applies for "stopServerMetadata"